presto-main/src/main/java/com/facebook/presto/metadata/InternalNode.java

```diff
@@ -66,24 +66,32 @@ public int getStatusCode()
     private final boolean coordinator;
     private final boolean resourceManager;
     private final NodeStatus nodeStatus;
+    //the pool that this node belongs to
+    private final String pool;
```

Comment:
dumb question : can we move nodes from one pool to another?

Reply:
It's a good question. the pool property is attached to a node similar to environment, node_id, data_location etc. So technically, moving nodes from one pool to another is possible by modifying node properties and rebooting the node.